src/fsharp/CheckExpressions.fs

```diff
@@ -7773,7 +7773,7 @@ and Propagate cenv (overallTy: OverallTy) (env: TcEnv) tpenv (expr: ApplicableEx
                         if isAdjacent then
                             if IsIndexerType cenv.g cenv.amap expr.Type then
                                 if cenv.g.langVersion.IsExplicitlySpecifiedAs50OrBefore() then
-                                    error (NotAFunctionButIndexer(denv, overallTy.Commit, vName, mExpr, mArg, false))
+                                    error (NotAFunctionButIndexer(denv, expr.Type, vName, mExpr, mArg, false))
```

Comment:
The `overallTy.Commit` type is not resolved yet so gives me the error message `stdin(1,13): error FS0003: This is a value of type ''a'` instead of the correct one

Has any problem replacing the `overallTy.Commit` with `expr.Type` that I'm missing?